docs/compose-file.md

```diff
@@ -89,6 +90,30 @@ definition.
 
 ### build
 
+Configuration options that are applied at build time.
+
+In version 1 this must be given as a string representing the context.
+
+  build: .
+
+In version 2 this can alternatively be given as an object with extra options.
+
+  version: 2
+  services:
+    web:
+      build: .
+
+    version: 2
+    services:
+      web:
+        build:
+          context: .
+          dockerfile: Dockerfile-alternate
+          args:
+            buildno: 1
```

Comment:
We have #2622 to finalize this, but I think what we'd like to do is only include the v2 docs in this section, and have a separate section at the bottom for migration, and differences between the versions.